File: solutions/kotlin/hamming/1/src/main/kotlin/Hamming.kt
import java.lang.IllegalArgumentException

object Hamming {

    fun compute(leftStrand: String, rightStrand: String): Int {
        if (leftStrand.length != rightStrand.length)
            throw IllegalArgumentException("left and right strands must be of equal length");

        var hammingDistance:Int = 0;

        leftStrand.forEachIndexed { index, c ->
            if (c != rightStrand[index]) hammingDistance+=1;
        };

        return hammingDistance;
    }
}

File: solutions/lua/hamming/1/hamming.lua
local Hamming = {}

function Hamming.compute(a,b)
  if #a ~= #b then
    return -1
  end

  local hamming = 0

  for i = 1, #a do
    if string.sub(a, i, i) ~= string.sub(b, i, i) then
      hamming = hamming + 1
    end
  end

  return hamming

end

return Hamming
